### ISSUE TITLE
Empty batch request being run

### ISSUE THREAD

**Issue by felixarntz** [google/site-kit-wp#339]
## Bug Description

Under certain conditions, the Site Kit dashboard triggers an empty batch request. This doesn't cause any visual errors, but is unnecessary and results in the request response being a 400 (because empty batch requests are a "Bad Request"). If a potential batch request doesn't include any actual request, it shouldn't be made.

## Steps to reproduce

Leave your browser console tab open for all of the following to notice the error. Also make sure to delete your browser cache for the site before starting.

1. Setup Site Kit and access the dashboard for the first time. Now a regular, working batch request is being made.
2. Reload the dashboard. Now no requests are being made because all response data is already cached on the client.
3. Activate the PageSpeed Insights module.
4. Access the dashboard. You will see the 400 error in the console, caused by the empty batch request. It seems that this request for some reason is made because the new PageSpeed Insights batch request has a "priority" value of 10, which causes it to be made as a secondary request.
5. Access the dashboard again. Now again no requests are being made (everything is fine) because the PSI data is now cached as well.

## Screenshots

<img width="1003" alt="Screenshot 2019-07-18 at 18 48 41" src="https://user-images.githubusercontent.com/3531426/62030039-6ebc3a00-b1e4-11e9-8289-475e40757260.png">
The failing batch request (because it's empty).

![image](https://user-images.githubusercontent.com/1621608/64844299-08626e00-d610-11e9-8708-3fa9bda4f65d.png)
The no data request as shown in the debugger. `currentRequest` is empty, but is not caught by the early return because `remainingDatapoints` is not empty.

## Additional Context

This only happened in the above context, but I've tried it multiple times, and it _always_ happened under these circumstances. Probably there is some logic that tries to make the primary batch request when a secondary batch request is present, regardless of whether the primary request has data.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* If a batch request ends up having no inner data requests at all, it shouldn't be executed.
* Particularly, right after activating the PageSpeed Insights module, no empty batch request should be made (please check in your browser's dev tools, see above in "Steps to reproduce").

## Implementation Brief

The problem is that requests with a priority of 10 or higher are pushed into a secondary request, even if there are no other requests to be made in the primary/first batch.

* Use `dataRequest.find` before requests are split into `deferredRequests` to check if there is any request with a priority less than 10
    - Add the result (found/not) to this conditional `! secondaryRequest && 10 <= request.priority` so that requests are only deferred if there are any primary requests.

## Changelog entry

* Fix empty batch request being triggered after activating the PageSpeed Insights module.

**Comment by felixarntz** [comment]
IB ✅

**Comment by adamsilverstein** [comment]
QA passed for this ticket, moving to Approval.

### Testing procedure
* Follow the "Steps to Reproduce" from above.
* confirm in network panel only one request is done on step 4 with expected response.
* Note not seeing any empty requests running, no 400 error response in the console.

**Comment by marrrmarrr** [comment]
I see only one regular request being made after activating PSI as well.
